src/OpenCL.cpp

```diff
@@ -1093,11 +1097,14 @@ void OpenCL::initialize(const int channels, const std::vector<int> & gpus,
         throw std::runtime_error("Error building OpenCL kernels.");
     }
 
-    ensure_thread_initialized();
+    // dummy gpu number, putting bogus number so that we don't misuse it
+    OpenCLContext tdata(0xffff);
```

Comment:
I don't think you can misuse anything here given that OpenCLContext is no longer a thread local. Just make it 0 or whatever is the first in cfg_gpu. There's also nothing thread related any more here.

Reply:
I will still leave the comment that the thread number here is bogus, just to prevent anybody from being confused...